e2e/common/cli/debug_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build integration
+// +build integration
+
+// To enable compilation of this file in Goland, go to "Settings -> Go -> Vendoring & Build Tags -> Custom Tags" and add "integration"
+
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	. "github.com/apache/camel-k/e2e/support"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestKamelCLIDebug(t *testing.T) {
+	WithNewTestNamespace(t, func(ns string) {
+		Expect(Kamel("install", "-n", ns).Execute()).To(Succeed())
+
+		t.Run("debug local default port check", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/yaml.yaml").Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, "yaml"), TestTimeoutLong).Should(Equal(v1.PodRunning))
+			Expect(portIsInUse("127.0.0.1", "5005")()).To(BeFalse())
+
+			go Kamel("debug", "yaml", "-n", ns).Execute()
```

Comment:
I wonder if the goroutine way is the best approach. I think that `kamel debug` execution will be hanging indefinitely.

Reply:
@squakez good point. It should be fixed with 
```
			debugTestContext, cancel := context.WithCancel(TestContext)
			defer cancel()
			go KamelWithContext(debugTestContext, "debug", "yaml", "-n", ns).Execute()
```